perf(search): only query DB for ft min max once

hypeJunction opened the pull request:
Reduces the DB queries for ft_min_word_len and ft_max_word_len by storing the
values in config

Fixes #6707

beck24 commented:
LGTM

mrclay commented on mod/search/start.php:
```diff
@@ -524,3 +512,41 @@ function search_exclude_robots($hook, $type, $text) {
+			// some servers don't have these values set which leads to exception
+			// we ignore the exception
+		}
+		if ($result) {
```
If the exception was thrown, $result won't be defined so this will cause a notice. Maybe move the assignments to the try and catch blocks?

mrclay replied:
Or just change to `if (!empty($result)) {`

hypeJunction committed:
perf(search): only query DB for fulltext min and max word length once

Reduces DB queries for fulltext min and max word length values by storing them in the config

Fixes #6707

hypeJunction commented:
@mrclay This should be better

beck24 commented:
LGTM again

jdalsem commented on mod/search/start.php:
```diff
@@ -524,3 +512,37 @@ function search_exclude_robots($hook, $type, $text) {
+	$min = (int) elgg_get_config('search_ft_min_word_len');
+	$max = (int) elgg_get_config('search_ft_max_word_len');
+
+	if (!$min || !$max) {
```
if you have min word length set to 0 this will always go... maybe not cast to int when fetching from config and check for null

hypeJunction replied:
I was thinking that but you can't have a zero length word
On Apr 11, 2015 12:10 PM, "Jeroen Dalsem" <notifications@github.com> wrote:

> In mod/search/start.php
> <https://github.com/Elgg/Elgg/pull/8165#discussion_r28193923>:
>
> > @@ -524,3 +512,37 @@ function search_exclude_robots($hook, $type, $text) {
> >
> >  	return $text;
> >  }
> > +
> > +/**
> > + * Returns minimum and maximum lengths of words for MySQL search
> > + * This function looks for stored config values, and, if none set,
> > + * queries the DB and saves them
> > + * @return stdClass An object with min and max properties
> > + */
> > +function search_get_ft_min_max() {
> > +
> > +	$min = (int) elgg_get_config('search_ft_min_word_len');
> > +	$max = (int) elgg_get_config('search_ft_max_word_len');
> > +
> > +	if (!$min || !$max) {
>
> if you have min word length set to 0 this will always go... maybe not cast
> to int when fetching from config and check for null
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8165/files#r28193923>.
>

jdalsem replied:
ah yes i see https://dev.mysql.com/doc/refman/5.1/en/server-system-variables.html#sysvar_ft_min_word_len
